docs/developers/evm-tracing/live-tracing.md

```diff
@@ -0,0 +1,304 @@
+	loggerOutput := &lumberjack.Logger{
+		Filename: filepath.Join(config.Path, "supply.jsonl"),
+	}
+
+	if config.MaxSize > 0 {
+		loggerOutput.MaxSize = config.MaxSize
+	}
+
+	logger := log.New(loggerOutput, "", 0)
+
+	supplyInfo := newSupplyInfo()
+
+	t := &Supply{
+		delta:  supplyInfo,
+		logger: logger,
+	}
+	return &tracing.Hooks{
+		OnBlockStart:    t.OnBlockStart,
+		OnBlockEnd:      t.OnBlockEnd,
+		OnGenesisBlock:  t.OnGenesisBlock,
+		OnTxStart:       t.OnTxStart,
+		OnBalanceChange: t.OnBalanceChange,
+		OnEnter:         t.OnEnter,
+		OnExit:          t.OnExit,
+	}, nil
+}
+
+func (s *Supply) resetDelta() {
+	s.delta = newSupplyInfo()
+}
+
+func (s *Supply) OnBlockStart(ev tracing.BlockEvent) {
+	s.resetDelta()
+
+	s.delta.Number = ev.Block.NumberU64()
+	s.delta.Hash = ev.Block.Hash()
+	s.delta.ParentHash = ev.Block.ParentHash()
+
+	// Calculate Burn for this block
+	if ev.Block.BaseFee() != nil {
+		burn := new(big.Int).Mul(new(big.Int).SetUint64(ev.Block.GasUsed()), ev.Block.BaseFee())
+		s.delta.burn(burn)
+	}
+	// Blob burnt gas
+	if blobGas := ev.Block.BlobGasUsed(); blobGas != nil && *blobGas > 0 && ev.Block.ExcessBlobGas() != nil {
+		var (
+			excess  = *ev.Block.ExcessBlobGas()
+			baseFee = eip4844.CalcBlobFee(excess)
+			burn    = new(big.Int).Mul(new(big.Int).SetUint64(*blobGas), baseFee)
+		)
+		s.delta.burn(burn)
+	}
+}
+
+func (s *Supply) OnBlockEnd(err error) {
+	out, _ := json.Marshal(s.delta)
+	s.logger.Println(string(out))
+}
+
+func (s *Supply) OnGenesisBlock(b *types.Block, alloc types.GenesisAlloc) {
+	s.resetDelta()
+
+	s.delta.Number = b.NumberU64()
+	s.delta.Hash = b.Hash()
+	s.delta.ParentHash = b.ParentHash()
+
+	// Initialize supply with total allocation in genesis block
+	for _, account := range alloc {
+		s.delta.Delta.Add(s.delta.Delta, account.Balance)
+	}
+
+	out, _ := json.Marshal(s.delta)
+	s.logger.Println(string(out))
+}
+
+func (s *Supply) OnBalanceChange(a common.Address, prevBalance, newBalance *big.Int, reason tracing.BalanceChangeReason) {
+	diff := new(big.Int).Sub(newBalance, prevBalance)
+
+	// NOTE: don't handle "BalanceIncreaseGenesisBalance" because it is handled in OnGenesisBlock
+	switch reason {
+	case tracing.BalanceIncreaseRewardMineUncle:
+	case tracing.BalanceIncreaseRewardMineBlock:
+		s.delta.Reward.Add(s.delta.Reward, diff)
+	case tracing.BalanceIncreaseWithdrawal:
+		s.delta.Withdrawals.Add(s.delta.Withdrawals, diff)
+	case tracing.BalanceDecreaseSelfdestructBurn:
+		// BalanceDecreaseSelfdestructBurn is non-reversible as it happens
+		// at the end of the transaction.
+		s.delta.Burn.Sub(s.delta.Burn, diff)
+	default:
+		return
+	}
+
+	s.delta.Delta.Add(s.delta.Delta, diff)
+}
+
+func (s *Supply) OnTxStart(vm *tracing.VMContext, tx *types.Transaction, from common.Address) {
+	s.txCallstack = make([]supplyTxCallstack, 0, 1)
+}
+
+// interalTxsHandler handles internal transactions burned amount
+func (s *Supply) interalTxsHandler(call *supplyTxCallstack) {
```

Comment:
Small typo, I think it might there is the "real" supply live tracer.

Reply:
Yep good catch will fix there too